packages/apple-pay-integration/src/create-apple-pay-payment-strategy.ts

```diff
@@ -12,11 +17,16 @@ const createApplePayPaymentStrategy: PaymentStrategyFactory<ApplePayPaymentStrat
     paymentIntegrationService,
 ) => {
     const { getHost } = paymentIntegrationService.getState();
+    const hostWindow = window;
 
     return new ApplePayPaymentStrategy(
         createRequestSender({ host: getHost() }),
         paymentIntegrationService,
         new ApplePaySessionFactory(),
+        new BraintreeIntegrationService(
+            new BraintreeScriptLoader(getScriptLoader(), hostWindow),
+            hostWindow,
+        ),
```

Comment:
I can see `BraintreeIntegrationService` is instaniated to support `getDeviceData()` only.
Does it make sense to create a `BraintreeIntegrationService` only when the gateway is `braintree`?

Reply:
do you mean something like this https://github.com/bigcommerce/checkout-sdk-js/pull/2131#discussion_r1305002849 ?

Reply:
Yes, I do.